pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -382,18 +389,23 @@ def on_validation_end(self, trainer, pl_module):
             elif self.check_monitor_top_k(current):
                 self._do_check_save(filepath, current, epoch, trainer, pl_module)
             elif self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: {self.monitor}  was not in top {self.save_top_k}')
+                log.info(f'Epoch {epoch:d}: {self.monitor} was not in top {self.save_top_k}')
 
         else:
             if self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: saving model to {filepath}')
+                log.info(f'Epoch {epoch:d}: saving model to {filepath}')
 
             assert trainer.global_rank == 0, 'tried to make a checkpoint from non global_rank=0'
             self._save_model(filepath, trainer, pl_module)
 
         if self.save_last:
-            filepath = os.path.join(self.dirpath, self.prefix + ModelCheckpoint.CHECKPOINT_NAME_LAST)
+            filename = self._format_checkpoint_name(
+                self.CHECKPOINT_NAME_LAST, epoch, ckpt_name_metrics, prefix=self.prefix
+            )
+            filepath = os.path.join(self.dirpath, f'{filename}{self.CHECKPOINT_SUFFIX}')
+            self._del_model(self.last_model_path)
             self._save_model(filepath, trainer, pl_module)
+            self.last_model_path = filepath
```

Comment:
can we (in a new PR pls) first rename the model to "oldname.ckpt.backup" before saving the model, and then delete the backup file after save has completed? Otherwise, if a crash happens during save_model, we will have lost the old checkpoint and have a corrupted file for the current.

Reply:
A new PR based on master or this branch?

Reply:
yes, just a small suggestion after this PR is merged. but it will touch other locations as well, for example topk saving.